### PR TITLE
pipeline: fix period frame size calculation

### PR DESCRIPTION
**Pull request by juimonen**
The period frame size calculation has issues by doing division instead
of multiplication. So fix this by introducing new function for buffer
period frames calculation where we multiply samplerate and
schedule_period and divide by 1000000. Also round up the result.

Signed-off-by: Jaska Uimonen <jaska.uimonen@intel.com>

### CODE REVIEW

File: src/audio/pipeline.c [hunk 1]
```diff
@@ -246,6 +246,25 @@ int pipeline_free(struct pipeline *p)
...
+	if (current->output_rate)
+		samplerate = current->output_rate;
+	else
+		samplerate = current->params.rate;
+
+	/* Samplerate is in kHz and period in microseconds.
+	 * As we don't have floats use scale divider 1000000.
+	 * Also integer round up the result.
+	 */
+	return ceil_divide(samplerate * period, 1000000);
```
Comment (tlauda) [inline src/audio/pipeline.c]: #include <sof/math/numbers.h>

Comment (juimonen) [reply src/audio/pipeline.c]: I thinks it is coming from component.h... so do you want me to include it explicitly to pipeline.h?

Comment (juimonen) [reply src/audio/pipeline.c]: sorry pipeline.c...

Comment (tlauda) [reply src/audio/pipeline.c]: Yes, the best practice is to include headers explicitly, so you won't be suprised during refactor.

Comment (juimonen) [reply src/audio/pipeline.c]: ok, got it and updated.

**Comment by wwittbrx** [comment]
I can confirm that this PR fixes bug #1772.

**Pull request merged**